dotfyle

```diff
@@ -158,9 +158,56 @@ update() {
   location="$HOME/.config/$dir_name"
   repos=$(find_git "$location")
   for repo in $repos; do
-    echo "Updating $repo"
-    git -C "$location/$repo" pull >/dev/null
-    #TODO: Better handle the --depth=1 issues
+    g="git -C $location/$repo"
+    changes=$($g diff --name-only)
+    if [ -z "$changes" ]; then
+      echo "Updating $repo"
+      $g pull >/dev/null
+    else
+      echo ""
+      echo "Cannot update $repo, found the following changes:"
+      echo "$changes"
+      while [ -z $done ]; do
+        if [ -v 1 ]; then
+          soi="$1"
+          soi=${soi#-}
+          soi=${soi#-}
+        else
+          echo "Would you like to stash, overwrite, or ignore? [soI]"
+          read -r soi
+        fi
+        case "$soi" in
+          "s"|"stash")
+            echo "Stashing your changes"
+            echo "you can find them at:"
+            echo "$location/$repo"
+            $g stash >/dev/null
+            $g pull >/dev/null
+            done=true
```

Comment:
You could consider popping the changes here if you want

Reply:
I could change the options to keep, overwrite, and skip. Skip makes more sense than ignore but I didn't want two s options. Keep would be stash then pop